e2e_tests/tests/test_sdk.py

```diff
@@ -474,3 +464,32 @@ def test_rp_workspace_mapping() -> None:
     finally:
         for wid in workspace_ids:
             bindings.delete_DeleteWorkspace(session=sess, id=wid)
+
+
+@pytest.mark.e2e_cpu
+def test_create_experiment_w_template(create_empty_dir) -> None:
+    sess = api_utils.user_session()
+    detobj = client.Determined._from_session(sess)
+    template_name = "test_template"
+    try:
+        # create template
+        tpl = bindings.v1Template(
+            name=template_name,
+            config=conf.load_config(conf.fixtures_path("templates/template.yaml")),
+            workspaceId=1,
+        )
+        tpl_resp = bindings.post_PostTemplate(sess, body=tpl, template_name=tpl.name)
+        # create experiment with template
+        with open(conf.fixtures_path("no_op/single-one-short-step.yaml")) as f:
+            config = util.yaml_safe_load(f)
+        emptydir = create_empty_dir
+        model_def = conf.fixtures_path("no_op/model_def.py")
+        exp = detobj.create_experiment(
+            config, emptydir, includes=[model_def], template=tpl_resp.template.name
+        )
+        exp.await_first_trial()
+
+        trials = exp.get_trials()
+        assert len(trials) == 1, trials
```

Comment:
My reading of this exact test is:

1. When creating an experiment with a minimal template, a trial is started before pytest times this test out
2. very soon after the first trial is started, the executed experiment has exactly one trial

I think I'd like a short chat about what exactly we should test. Should we test that the system can start an experiment with _any_ template? Should we test that it actually uses config from the template?

I don't know! But I'd like to work out what to test and what assertions and names those tests should have.